File: ts-sdk/src/sdk/models/operations/postapidisplayads.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";


export enum PostApiDisplayAdsRequestBodyDisplayToEnum {
    All = "all",
    LoggedIn = "logged_in",
    LoggedOut = "logged_out"
}

export enum PostApiDisplayAdsRequestBodyPlacementAreaEnum {
    SidebarLeft = "sidebar_left",
    SidebarLeft2 = "sidebar_left_2",
    SidebarRight = "sidebar_right",
    PostSidebar = "post_sidebar",
    PostComments = "post_comments"
}


export class PostApiDisplayAdsRequestBody extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=approved" })
  approved?: boolean;

  @SpeakeasyMetadata({ data: "json, name=body_markdown" })
  bodyMarkdown: string;

  @SpeakeasyMetadata({ data: "json, name=display_to" })
  displayTo?: PostApiDisplayAdsRequestBodyDisplayToEnum;

  @SpeakeasyMetadata({ data: "json, name=name" })
  name: string;

  @SpeakeasyMetadata({ data: "json, name=organization_id" })
  organizationId?: number;

  @SpeakeasyMetadata({ data: "json, name=placement_area" })
  placementArea: PostApiDisplayAdsRequestBodyPlacementAreaEnum;

  @SpeakeasyMetadata({ data: "json, name=published" })
  published?: boolean;

  @SpeakeasyMetadata({ data: "json, name=tag_list" })
  tagList?: string;
}


export class PostApiDisplayAdsRequest extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "request, media_type=application/json" })
  request?: PostApiDisplayAdsRequestBody;
}


export class PostApiDisplayAdsResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;
}

File: python-sdk/src/foremapi/models/operations/post_api_display_ads.py
import dataclasses
from typing import Optional
from enum import Enum
from dataclasses_json import dataclass_json
from foremapi import utils

class PostAPIDisplayAdsRequestBodyDisplayToEnum(str, Enum):
    ALL = "all"
    LOGGED_IN = "logged_in"
    LOGGED_OUT = "logged_out"

class PostAPIDisplayAdsRequestBodyPlacementAreaEnum(str, Enum):
    SIDEBAR_LEFT = "sidebar_left"
    SIDEBAR_LEFT_2 = "sidebar_left_2"
    SIDEBAR_RIGHT = "sidebar_right"
    POST_SIDEBAR = "post_sidebar"
    POST_COMMENTS = "post_comments"


@dataclass_json
@dataclasses.dataclass
class PostAPIDisplayAdsRequestBody:
    body_markdown: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('body_markdown') }})
    name: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('name') }})
    placement_area: PostAPIDisplayAdsRequestBodyPlacementAreaEnum = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('placement_area') }})
    approved: Optional[bool] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('approved') }})
    display_to: Optional[PostAPIDisplayAdsRequestBodyDisplayToEnum] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('display_to') }})
    organization_id: Optional[int] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('organization_id') }})
    published: Optional[bool] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('published') }})
    tag_list: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('tag_list') }})
    

@dataclasses.dataclass
class PostAPIDisplayAdsRequest:
    request: Optional[PostAPIDisplayAdsRequestBody] = dataclasses.field(default=None, metadata={'request': { 'media_type': 'application/json' }})
    

@dataclasses.dataclass
class PostAPIDisplayAdsResponse:
    content_type: str = dataclasses.field()
    status_code: int = dataclasses.field()
    
